core/src/com/g16/feyrune/model/handler/BattleHandler.java

```diff
@@ -4,75 +4,110 @@
+        int attackIndex = (int) (Math.random() * attacks.length);
+        BaseAttack currentAttack = attacks[attackIndex];
+        useAttack(friend, enemy, currentAttack);
     }
 
-    private BaseAttack chooseMove() {
-        System.out.println("choose move");
-        String chosenMove = "attackname"; //replace with the listener from the UI
-        BaseAttack chosenAttacker = null; // don't know how to find what move is choosen
-        return chosenAttacker;
+    /**
+     * Chooses an attack for the enemy
+     */
+    private void enemyChooseAttack() {
+        BaseAttack[] enemyAttacks = enemy.getAttacks();
+        int attackIndex = (int) (Math.random() * enemyAttacks.length);
+        BaseAttack currentAttack = enemyAttacks[attackIndex];
+        useAttack(enemy, friend, currentAttack);
     }
 
-    private void battleLoop(BaseCreature friend, BaseCreature enemy) {
-        /**
-         * This method, and accompanying AttackHandler should probably be rewritten
-         * to use a more generic battle loop with as little logic as possible.
-         * 1. Get next
-         * 2. Get move
-         * 3. Execute move
-         */
-        Pair<BaseCreature, BaseCreature> order = getOrderFromSpeed(friend, enemy);
-        BaseCreature fastest = order.fst;
-        BaseCreature slowest = order.snd;
-
-        int loop = 0;
-        while (!combatIsActive) {
-            BaseAttack playerMove = chooseMove();
-            BaseAttack enemyMove = null; //random move
-
-            if (fastest.isFriend()) {
-                AttackHandler.handleAttack(fastest, slowest, playerMove);
-                if (!combatIsActive) break;
-
-                AttackHandler.handleAttack(slowest, fastest, enemyMove);
-            } else {
-                AttackHandler.handleAttack(fastest, slowest, enemyMove);
-                if (!combatIsActive) break;
-                AttackHandler.handleAttack(slowest, fastest, playerMove);
-            }
-        }
+    /**
+     * Controls which creature should attack and makes them choose one
+     */
+    private void chooseAttack() {
+        BaseCreature currentMonster = turnOrder.remove(0);
+        if (currentMonster.isFriend()) friendChooseAttack();
+        else enemyChooseAttack();
+        generateAttackOrder();
+    }
 
+    /**
+     * Uses an attack through the attack handler
+     *
+     * @param attacker:    the creature that is attacking
+     * @param defender:    the creature that is being attacked
+     * @param attackToUse: the attack that is being used
+     */
+    private void useAttack(BaseCreature attacker, BaseCreature defender, BaseAttack attackToUse) {
+        AttackHandler.handleAttack(attacker, defender, attackToUse);
     }
 
-    public static void exitBattle() {
+    /**
+     * Checks if the battle is over, if not, continues the battle
+     */
+    private void battleLoop() {
+        while (combatIsActive) {
+            chooseAttack();
```

Comment:
This method name is slightly misleading, as it seems like all you do is choosing (getting) the attack, but you actually choose and execute the attack. Either it needs a better name, or it breaks the Command-Query Principle